### DISCUSSION TITLE
[feature request] Add id to html elements so that we can link to them in the url

### DISCUSSION THREAD

**Discussion by sbalay** [jy95/docusaurus-json-schema-plugin#120] [Ideas]
It would be nice if we could use `#` in the url to link directly to specific fields in the json schema representation

**Comment by jy95** [comment]
Hello @sbalay,

Welcome - the anchor idea was something I started reflexion but I may be biased - I'm open for discussion .
In [JSON Schema specs](https://json-schema.org/draft/2020-12/json-schema-core.html#name-base-uri-anchors-and-derefe), there are multiple ways to achieve what you ask. 

In my personal opinion, [Json-Pointer](https://json-schema.org/understanding-json-schema/structuring.html#json-pointer) as id seems the wise choice : 
1. You don't have to modify your JSON Schema (like putting `$anchor` for example)
2. It should support nested paths
3. Little trade-off are the [encoding rules](https://datatracker.ietf.org/doc/html/rfc6901#section-3) for the id in urls but it is better than nothing

Let's take some examples to illustrate my thinking : 

### Object properties

Given this JSON Schema : 
```json
{
  "$id": "https://example.com/schemas/address",
  "type": "object",
  "properties": {
    "street_address":
      { "type": "string" },
    "city": { "type": "string" },
    "state": { "type": "string" }
  },
  "required": ["street_address", "city", "state"]
}
```

I would like to target the `street_address` property so JSON Pointer is `/properties/street_address`
Url would be something like `/something#/properties/street_address`

### Array Properties
Given this JSON Schema : 
```json
{
  "type": "array",
  "prefixItems": [
    { "type": "number" },
    { "type": "string" },
    { "enum": ["Street", "Avenue", "Boulevard"] },
    { "enum": ["NW", "NE", "SW", "SE"] }
  ]
}
```

I would like to target the 3rd entry of the array so JSON Pointer is `/prefixItems/3`
Url would be something like `/something#/prefixItems/3`

### Example of escaped attribute
Given this JSON Schema : 
```json
{
  "$id": "https://example.com/schemas/address",
  "type": "object",
  "properties": {
    "street_address":
      { "type": "string" },
    "city": { "type": "string" },
    "country/state": { "type": "string" }
  },
  "required": ["street_address", "city", "state"]
}
```

I would like to target the `country/state` property so JSON Pointer is `/properties/country~1state`
Because of [encoding rules](https://datatracker.ietf.org/doc/html/rfc6901#section-3), url would be something like `/something#/properties/country~1state`

I'm curious on how Docusaurus `Collapsible` component behalves for nested levels but I guess it is worthy an try ...

Feel free to share your thoughts ;)

**Comment by jy95** [comment]
@sbalay https://github.com/jy95/docusaurus-json-schema-plugin/pull/234 should solve your issue with fields - could you give a look with the preview playground before I release the 1.10.0 version ?

**Comment by sbalay** [comment]
Hey @jy95, thanks for the hard work!

I played around with the [deploy preview](https://deploy-preview-234--delicate-torrone-5d35ee.netlify.app/docs/demo/viewer) and it looks really good!

**Reply by jy95** [reply]
@sbalay 1.10.0 is out : have fun ;)
